### ISSUE TITLE
SCIM API tends to miss-manage date/time attributes when custom SCIM scripts are also involved

### ISSUE THREAD

**Issue by aliaksander-samuseu** [GluuFederation/scim#70]
[test_scim_script.zip](https://github.com/GluuFederation/scim/files/11492739/test_scim_script.zip)
## Description
As it was found out during recent troubleshooting session involving SCIM custom scripts that had to persist attributes on their own (in addition to the ones passed in the request) - at least in user creation flow - if attribute of type "Date" is persisted from inside the script, it leads to varying but unexpected results, which may depend on the type of persistence layer (specifically, differences were observed when using LDAP vs MySQL).

## Preconditions
1. Ideally, two Gluu Servers (CE) are needed, one using LDAP, the other one using MySQL for persistence; if only one is option, it should be MySQL-based
2. Deployed test custom SCIM script attached above
3. Several test attributes of type "Date" set in oxTrust, with a compatible type used for corresponding DB attribute (`1.3.6.1.4.1.1466.115.121.1.24` type for LDAP and "datetime()" type for MySQL); the attached script expects that "activatedAt" and "lastLoginAt" attributes are existing; there also should be at least one more date/time attribute which will be passed in SCIM request explicitly; all the attributes need to be added to SCIM extended schema (by checking corresponding checkbox in their properties)

## Steps to reproduce
1. Send a user creation request which includes date/time attribute that is not the one of the two set inside the script
2. Check the API's response, contents of attributes' DB entries and scim logs

## Results

In MySQL none of the attributes set from inside the script are present in immediate user creation response from API, only the one sent in the request makes it there; it will appear in subsequent GET request for that user entry, and is pushed to DB, so it's not a persistence issue.

In case of LDAP it behaves a bit different - if it's a string what is passed in "strTime" variable in a line like below:
```
user.setCustomAttribute('lastLoginAt', strTime)
```
...then it will appear in response. At the other hand, if strTime contains a "java.util.Date" object (which is, according to @jgomer2001 , is the most appropriate and DB-independent approach), such attribute won't appear in immediate response. The rest seems to stay the same.

Also, when textual representation of the generalized time is used when passing the value, it seems like OpenDJ and MySQL react to it differently. Using the approach suggested by Jose, as below:
```
        objTime = Date() # this is a full date with milliseconds and so on
        strTime = StaticUtils.encodeGeneralizedTime(objTime)
        user.setCustomAttribute('activatedAt', strTime)
```
..the date/time string produced will look like this: `20230411190818.574Z`. OpenDJ consumes such value without complaints, while MySQL rejects it because of the trailing "Z" character. It can still be fixed with line like below:
```
        strTime = strTime.rstrip("Zz")
```
..but adds to confusion and makes scim scripts less portable

## Expected results

Ideally, we need our custom scripts to be DB-agnostic to make it possible to quickly share and deploy them between different environments. If possible, we need to find a way to handle date/time attributes uniformly in these two, and other possible cases. Passing (and manipulating) a java time object seems like a solid idea, but the issue with it missing from API response needs to be fixed first.

**Comment by jgomer2001** [comment]
> In MySQL none of the attributes set from inside the script are present in immediate user creation response from API, only the one sent in the request makes it there; 

Cannot reproduce. Just install 4.4.2 and it returns both: the attributes in the payload plus the ones set in the cust script

**Comment by jgomer2001** [comment]
Fixed applied for getting java.util.Date instances in the response payload when using LDAP as DB.

Fix commited to 4.5.2 branch - not master (there won' be a 4.6.x release apparently)
